src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+
+final class OptionConfigurator
+{
+    private $name;
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed values for this option.
+     *
+     * @param mixed ...$allowedValues One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function addAllowedValues(...$allowedValues): self
```

Comment:
do we actually need the "adders" here?

Reply:
> do we actually need/want the "adders" here?

I'd say yes, that's the way we can add more types/values/normalizers to existent options that *were* defined e.g. into a parent form type via `configureOptions()`.

> id argue the configurator pattern should be used for mutating existing options

What do you have in mind exactly? Note that `->define()` method is already used for that.

Reply:
using `define()` somewhat implies we define a "new option", so the "adder" implies we'd _redefine_ something

no real issue for me :) other then using "define" for "re-defining" may look weird.

Reply:
Yes, I got it. Maybe `set()` is a better name?

Reply:
Or maybe remove the adders and throw on `define()` if the option already exists, wdyt?